metabase/metabase
# Enable bug reporting

Closes: 

https://github.com/metabase/metabase/issues/49585
https://github.com/metabase/metabase/issues/49586
https://github.com/metabase/metabase/issues/49664
https://github.com/metabase/metabase/issues/50273
https://github.com/metabase/metabase/issues/50732

### Description

Since the feedback loop on the new bug reporting modal got real messy real fast, I decided to group all the changes under one branch. This PR contains changes from following PRs:

https://github.com/metabase/metabase/pull/50740
https://github.com/metabase/metabase/pull/50733
https://github.com/metabase/metabase/pull/50306
https://github.com/metabase/metabase/pull/50277
https://github.com/metabase/metabase/pull/49954

# Main features
ability to open diagnostic window [using Profile settings menu and command palette](https://github.com/metabase/metabase/pull/50856/commits/19427d4bf2ad2ffdc7aef88456de948491866506):
![Screenshot 2024-12-04 at 11 42 48](https://github.com/user-attachments/assets/b300f4cf-7751-4921-9fc6-ae45610c5058)
![Screenshot 2024-12-04 at 11 48 17](https://github.com/user-attachments/assets/7caef175-acda-4363-a085-3b1a6bc5a449)

[Rewrite of the Diagnostic modal](https://github.com/metabase/metabase/pull/50856/commits/c3e99f66c51e6492e3214f6dc3a8b685a4b77fdb), splitting it into two new components. One for downloading diagnostic data as we know it (with a facelift) and another one for submitting bugs to Slack. In future, we’ll be sending bugs elsewhere. Right now we enable sending them to slack as part of internal testing. Details on the whole project can be [found here](https://www.notion.so/metabase/Make-it-blazingly-fast-for-us-to-reproduce-and-fix-bugs-11f69354c90180418a27cc39714f3047?pvs=4)

![Screenshot 2024-11-20 at 15 10 30](https://github.com/user-attachments/assets/d7d7259e-8b5f-4653-a9ff-28f04f42e14c)
![Screenshot 2024-11-20 at 15 10 27](https://github.com/user-attachments/assets/2ffee9e1-3b03-41ad-b4ec-73a796f8d15a)
![Screenshot 2024-11-20 at 15 09 18](https://github.com/user-attachments/assets/0e60a35e-4b99-4361-a73f-1c9fdc184f02)

[Analytics](https://github.com/metabase/metabase/pull/50856/commits/e8016e92031c3061887214dc55ec1c9aa15a54f2) for whenever diagnostic modal is opened. (cc @iethree I implemented [your suggestions](https://github.com/metabase/metabase/pull/50306#discussion_r1868300141))

[Collecting basic browser and os information](https://github.com/metabase/metabase/pull/50856/commits/120f44cfa2b2ce1952a20b0655ac3c94e2852059) into diagnostics.

[Ability to jump from slack report message](https://github.com/metabase/metabase/pull/50856/commits/dd670685bccf06101befe934e2ddc11341a37496) right into debugger

### How to verify

1. Press cmd + k to open command palette and search for "file a bug" or use cmd + f1 to open the diagnostic command directly. 
2. download diagnostic data (now contains browser informarion
3. to enable bug reporting to slack, make sure to run your metabase instance with `MB_ENABLE_BUG_REPORTING` env var set to true
4. in slack, click "Jump to debugger"

### Checklist
